resteasy-core/src/main/java/org/jboss/resteasy/specimpl/ResteasyUriInfo.java

```diff
@@ -499,4 +539,97 @@ public URI relativize(URI uri)
+            if (absoluteUri.charAt(pathIdx - 1) == ':' && absoluteUri.charAt(pathIdx + 1) == '/') {
+               pathIdx = pathIdx + 2;
+               int tmp = absoluteUri.indexOf('/', pathIdx);
+               if (tmp > -1) pathIdx = tmp;
+            }
+         }
+         queryIdx = pathIdx > -1 ? absoluteUri.indexOf('?', pathIdx) : absoluteUri.indexOf('?');
+         int endPath = queryIdx > -1 ? queryIdx : absoluteUri.length();
+         pathStart = pathIdx > -1 ? pathIdx : 0;
+
+         String tmpEncodedPath = pathStart >= 0 && endPath > pathStart ? absoluteUri.substring(pathStart, endPath) : "";
+         tmpEncodedPath = PathHelper.getEncodedPathInfo(tmpEncodedPath, contextPath);
+         if (tmpEncodedPath.length() == 0 || tmpEncodedPath.charAt(0) != '/')
+         {
+            tmpEncodedPath = "/" + tmpEncodedPath;
+         }
+         encodedPath = tmpEncodedPath;
+
+         path = Encode.decodePath(encodedPath);
+         ProcessPathResult processPathResult = doProcessPath(encodedPath);
+         this.encodedPathSegments = processPathResult.getEncodedPathSegments();
+         this.pathSegments = processPathResult.getPathSegments();
+         this.matchingPath = processPathResult.getMatchingPath();
+
+         // ensure that we don't include any of the query params into the cache key
+         // since they don't affect the production of data at all
+         this.cacheKey = (queryIdx > -1 ? absoluteUri.substring(0, queryIdx) : absoluteUri) + "ResteasyUriInfo-Delimiter" + contextPath;
+      }
+
+      public int getQueryIdx() {
+         return queryIdx;
+      }
+
+      public int getPathStart() {
+         return pathStart;
+      }
+
+      public String getEncodedPath() {
+         return encodedPath;
+      }
+
+      public String getPath() {
+         return path;
+      }
+
+      public List<PathSegment> getEncodedPathSegments() {
+         return encodedPathSegments;
+      }
+
+      public List<PathSegment> getPathSegments() {
+         return pathSegments;
+      }
+
+      public String getMatchingPath() {
+         return matchingPath;
+      }
+
+      /**
+       * The InitData can be cached if it doesn't contain matrix variables which would result
+       * in a cache entry per different value of any matrix param
+       * Query params are not a problem since there names and values don't affect the production of InitData
+       */
+      public static boolean canBeCached(String absoluteUri) {
+         return !absoluteUri.contains(";");
+      }
+
+      // ensure that we don't include any of the query params into the cache key
+      // since they don't affect the production of data at all
+      public static String getCacheKey(String absoluteUri, String contextPath) {
+         int queryIdx = absoluteUri.lastIndexOf('?');
+         return (queryIdx > -1 ? absoluteUri.substring(0, queryIdx) : absoluteUri) + "ResteasyUriInfo-Delimiter" + contextPath;
+      }
+   }

```

Comment:
This method is not called anywhere in RESTEasy, so assuming it's needed in Quarkus (or other consumers of RESTEasy), we need a test covering for it. Moreover, line 631 has the same logic of line 587 and I believe it's important for those to stay in synch, which would suggest me to factor that into a single dedicated method.

Reply:
Nice catch! I got rid of the other part of the code since it wasn't being used anywhere.